### PR TITLE
feat: multi type chart support

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

add bar chart for funnel/event/path analysis

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 1]
```diff
@@ -719,9 +753,21 @@ export function getFunnelTableVisualRelatedDefs(viewName: string, colNames: stri
   return columnConfigurations;
 }
 
-export function getEventLineChartVisualDef(visualId: string, viewName: string, titleProps: DashboardTitleProps, groupColumn: string) : Visual {
+export function getEventChartVisualDef(visualId: string, viewName: string, titleProps: DashboardTitleProps,
+  quickSightChartType: QuickSightChartType, groupColumn: string) : Visual {
 
-  const visualDef = readFileSync(join(__dirname, './templates/event-line-chart.json'), 'utf8');
+  let templatePath = '';
+  if (quickSightChartType === QuickSightChartType.LINE) {
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:760]: how about like this,

```
templatePath = `./templates/event-${quickSightChartType}-chart.json`;
```

Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:760]: use `switch` is better

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts [hunk 2]
```diff
@@ -719,9 +753,21 @@ export function getFunnelTableVisualRelatedDefs(viewName: string, colNames: stri
...
+    templatePath = './templates/event-line-chart.json';
+  } else if (quickSightChartType === QuickSightChartType.BAR) {
+    templatePath = './templates/event-bar-chart.json';
+  } else {
+    const errorMessage = `unsupported quicksight char type ${quickSightChartType}`;
+    logger.debug(errorMessage);
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts:766]: change to `warn`

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts]: fixed

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts]: fixed

File: src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts]: After solving the above problem, we no longer need to use the `switch`

**Review by zxkane — APPROVED** [review]

**Pull request merged**